persistence/sql/persister_identity.go

```diff
@@ -202,6 +202,43 @@ func (p *Persister) createVerifiableAddresses(ctx context.Context, i *identity.I
 	return nil
 }
 
+func (p *Persister) updateRecoveryAddresses(ctx context.Context, i *identity.Identity) error {
```

Comment:
This probably needs some tests to ensure the diff is applied correctly :)

Reply:
Thanks, makes sense. I only tested for the actual bug for now, but I can extend this to all relationships.

Reply:
I just looked at the existing tests, and we already test that `UpdateIdentity` propagates updates to the recovery address here:

https://github.com/ory/kratos/blob/a059ef079ceffdec6364e174e782b171cbbdc25b/identity/test/pool.go#L769-L803

Reply:
👍 perfect